### PR TITLE
Make carousel indicators actual buttons (backport to v4)

### PR DESCRIPTION
**Pull request by patrickhlauke**
Manual backport of #32661

### CODE REVIEW

File: js/src/carousel.js
Comment (XhmikosR) [inline js/src/carousel.js:400]: Just use `parseInt` directly, no need for polyfills.

Comment (patrickhlauke) [reply js/src/carousel.js:400]: fair enough

**Review by mdo — APPROVED** [review]

**Comment by XhmikosR** [comment]
Need another approval just in case @rohit2sharma95.

**Review by rohit2sharma95 — APPROVED** [review]

**Comment by XhmikosR** [comment]
I marked it as draft; I'm not 100% sure about the CSS changes and specificity since people might override the selectors.

**Review by XhmikosR — CHANGES_REQUESTED** [review]
Temporarily blocking the PR

**Comment by patrickhlauke** [comment]
Setting this to ready for review, but worth deciding if we're going ahead with this backport or scrapping it.

**Comment by patrickhlauke** [comment]
@mdo @XhmikosR your call if this is too much of a breaking change or not. if we feel it is, suggest just closing the PR and not backporting.

**Comment by mdo** [comment]
Revisiting this, I think we will have to skip it for the breaking changes in those selectors. So sorry for the wasted effort so long ago! 😓

**Comment by patrickhlauke** [comment]
no worries, 5 is where it's at now anyway

**Pull request closed without merging**
